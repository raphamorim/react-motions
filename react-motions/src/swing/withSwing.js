import React from 'react'
import styles from './styles'

function withSwing(Component) {
  const style = {
    WebkitAnimationName: "swing",
    animationName: "swing",
    WebkitAnimationDuration: "1s",
    animationDuration: "1s",
    WebkitAnimationFillMode: "both",
    animationFillMode: "both",
    WebkitTransformOrigin: "top center",
    transformOrigin: "top center",
  }

  return (
    <div style={style}>
      {Component}
    </div>
  )
}

export default withSwing
