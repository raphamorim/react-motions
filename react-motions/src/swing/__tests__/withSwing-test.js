import React from 'react';
import withSwing from '../withSwing';

describe('withSwing', () => {
  it('should render', () => {
    const wrapper = shallow(withSwing(<div>Swing</div>));

    expect(wrapper).toMatchSnapshot();
  });
});
