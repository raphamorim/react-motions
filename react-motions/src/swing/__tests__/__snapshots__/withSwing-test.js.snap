// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSwing should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    style={
      Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "swing",
        "WebkitTransformOrigin": "top center",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "swing",
        "transformOrigin": "top center",
      }
    }
  >
    <div>
      Swing
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Swing
      </div>,
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "swing",
        "WebkitTransformOrigin": "top center",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "swing",
        "transformOrigin": "top center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Swing",
      },
      "ref": null,
      "rendered": "Swing",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Swing
        </div>,
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "swing",
          "WebkitTransformOrigin": "top center",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "swing",
          "transformOrigin": "top center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Swing",
        },
        "ref": null,
        "rendered": "Swing",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
